### PR TITLE
Major refactor: Package restructure

### PR DESCRIPTION
**Pull request by astubbs**
Description...

### Checklist

- [ ] Documentation (if applicable)
- [ ] Changelog

### CODE REVIEW

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/kafkabridge/Message.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package io.confluent.parallelconsumer.kafkabridge;
+
+/*-
+ * Copyright (C) 2020-2022 Confluent, Inc.
+ */
+
+public class Message {
```
Comment (astubbs) [inline parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/kafkabridge/Message.java:7]: docs

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/offsets/OffsetMapCodecManager.java [hunk 2]
```diff
@@ -186,14 +185,16 @@ String serialiseIncompleteOffsetMapToBase64(long finalOffsetForPartition, Partit
      * <p>
      * Can remove string encoding in favour of the boolean array for the `BitSet` if that's how things settle.
      */
-    byte[] encodeOffsetsCompressed(long finalOffsetForPartition, PartitionState<K, V> partitionState) throws NoEncodingPossibleException {
-        var incompleteOffsets = partitionState.getIncompleteOffsetsBelowHighestSucceeded();
-        long highestSucceeded = partitionState.getOffsetHighestSucceeded();
+    byte[] encodeOffsetsCompressed(long finalOffsetForPartition, HighestOffsetAndIncompletes partitionState) throws NoEncodingPossibleException {
+        var incompleteOffsets = partitionState.getIncompleteOffsets();
+        //long highestSucceeded = partitionState.getOffsetHighestSucceeded();
```
Comment (astubbs) [inline parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/offsets/OffsetMapCodecManager.java:190]: remove

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/sharedstate/CommitData.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import lombok.Value;
+import lombok.experimental.Delegate;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Value
```
Comment (astubbs) [inline parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/sharedstate/CommitData.java:15]: docs

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/sharedstate/CommitRequest.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+import lombok.Value;
+
+import java.util.UUID;
+
+/**
+ * Commit request message
+ */
+@Value
+public class CommitRequest {
```
Comment (astubbs) [inline parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/sharedstate/CommitRequest.java:15]: extend

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/sharedstate/CommitResponse.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+ * Copyright (C) 2020-2022 Confluent, Inc.
+ */
+
+import lombok.Value;
+
+/**
+ * Commit response message, linked to a {@link CommitRequest}
+ */
+@Value
+public class CommitResponse {
```
Comment (astubbs) [inline parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/sharedstate/CommitResponse.java:13]: extend

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/sharedstate/ControllerEventMessage.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+/**
+ * An inbound message to the controller.
+ * <p>
+ * Currently, an Either type class, representing either newly polled records to ingest, or a work result.
+ */
+@AllArgsConstructor
+@RequiredArgsConstructor(access = PRIVATE)
+@Getter
+@EqualsAndHashCode
+public class ControllerEventMessage<K, V> {
```
Comment (astubbs) [inline parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/sharedstate/ControllerEventMessage.java:26]: extend

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/sharedstate/PartitionEventMessage.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+ */
+
+
+import lombok.Value;
+import org.apache.kafka.common.TopicPartition;
+
+import java.util.Collection;
+
+@Value
+public class PartitionEventMessage {
```
Comment (astubbs) [inline parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/sharedstate/PartitionEventMessage.java:14]: extend

File: parallel-consumer-vertx/src/main/java/io/confluent/parallelconsumer/vertx/VertxParallelEoSStreamProcessor.java [hunk 8]
```diff
@@ -191,12 +191,12 @@ private void addVertxHooks(final PollContextInternal<K, V> context, final Future
             send.onSuccess(h -> {
                 log.debug("Vert.x Vertical success");
                 wc.onUserFunctionSuccess();
-                addToMailbox(wc);
+                super.bus.sendWorkResultEvent(wc);
```
Comment (astubbs) [inline parallel-consumer-vertx/src/main/java/io/confluent/parallelconsumer/vertx/VertxParallelEoSStreamProcessor.java:194]: implement delegate for add to mb

**Comment by eddyv** [comment]
Closing - Stale.

**Pull request closed without merging**